### PR TITLE
fix: users not joining org/team when signing up with API v2 invite tokens

### PR DESCRIPTION
**Pull request by anikdhabal**
## What does this PR do?

Fixes signup with API v2 invite-link tokens so new users are added to the correct org/team and the form behaves as expected.

### Bug Fixes
- Support non-email token identifiers from API v2 invite links (e.g., `invite-link-for-teamId-123`)
- Check for existing users and derive username only when the token identifier is a valid email; skip premium-username check otherwise
- Require tokens to be associated with a team; redirect to an error page if missing
- Prepopulate form fields only for email-based tokens; keep fields empty for invite-link tokens
- Auto-generate username for org invite links even when `orgAutoAcceptEmail` is not configured

### E2E Tests
- Rewrote E2E tests to follow the UI flow pattern (matching `organization-invitation.e2e.ts`)
- Tests now use `getInviteLink()` to get invite links from the UI instead of creating tokens directly in the database
- Added tests for: org invite link signup, sub-team invite link signup, and email-based token signup (regression test)

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] I have updated the developer docs in /docs if this PR makes changes that would require a documentation change. N/A - no docs changes needed.
- [x] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?

1. Create an organization and generate an invite link from the members page
2. Open the invite link in an incognito browser
3. Fill in email and password (username field should be hidden)
4. Submit the form - user should be created and added to the organization
5. Verify the user appears in the organization members list

## Human Review Checklist

- [ ] Verify the E2E test pattern matches existing tests in `organization-invitation.e2e.ts`
- [ ] Verify username auto-generation works when `orgAutoAcceptEmail` is null (uses `emailUser-emailDomain` format)
- [ ] Verify email-based token signup still works (regression test covers this)
- [ ] Check that the `prepopulateFormValues` condition change (`!isOrgInviteByLink && isValidEmail`) doesn't break any edge cases

## Checklist

- [x] I have read the [contributing guide](https://github.com/calcom/cal.com/blob/main/CONTRIBUTING.md)
- [x] My code follows the style guidelines of this project
- [x] I have checked if my changes generate no new warnings

---

**Link to Devin run:** https://app.devin.ai/sessions/30981c14c10b4d9abbe2dd429bf4d50b
**Requested by:** anik@cal.com (@anikdhabal)

### CODE REVIEW

File: apps/web/lib/signup/getServerSideProps.tsx [hunk 1]
```diff
@@ -125,45 +125,48 @@ export const getServerSideProps = async (ctx: GetServerSidePropsContext) => {
...
-        destination: `/auth/login?callbackUrl=${WEBAPP_URL}/${ctx.query.callbackUrl}`,
+        destination: `/auth/error?error=Verification Token is not associated with any team`,
       },
-    };
+    } as const;
+  }
+
+  const isValidEmail = checkValidEmail(verificationToken.identifier);
+  if (isValidEmail) {
+    const existingUser = await prisma.user.findFirst({
```
[the remark this answers is not in the document]
Comment (alishaz-polymath) [reply apps/web/lib/signup/getServerSideProps.tsx:139]: @anikdhabal We should definitely use explicit select

**Review by alishaz-polymath — COMMENTED** [review]
LGTM, except the missing explicit select.

File: apps/web/lib/signup/getServerSideProps.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (anikdhabal) [reply apps/web/lib/signup/getServerSideProps.tsx:139]: Done

**Review by alishaz-polymath — DISMISSED** [review]

**Comment by alishaz-polymath** [comment]
@anikdhabal Tests failing, related.

**Review by alishaz-polymath — APPROVED** [review]
LGTM
